### ISSUE TITLE
Messages without text cause errors in several situations

### ISSUE THREAD

**Issue by v-kydela** [Microsoft/botbuilder-dotnet#1211]
## Version
4.1.5

## Describe the bug
There are certain situations involving recognizers (such as when using a choice prompt) where the SDK is unable to respond gracefully to a message with a null or empty `Text` property. Such a message can be generated by a card action or by the user uploading an attachment for example. In these situations, the SDK will throw an error like this (I've abbreviated the stack trace):
```
System.ArgumentNullException: Value cannot be null.
Parameter name: utterance
   at Microsoft.Bot.Builder.Dialogs.Choices.Find.FindChoices(String utterance, IList`1 choices, FindChoicesOptions options) in D:\a\1\s\libraries\Microsoft.Bot.Builder.Dialogs\Choices\Find.cs:line 31
   at Microsoft.Bot.Builder.Dialogs.Choices.ChoiceRecognizers.RecognizeChoices(String utterance, IList`1 list, FindChoicesOptions options) in D:\a\1\s\libraries\Microsoft.Bot.Builder.Dialogs\Choices\ChoiceRecognizers.cs:line 25
   at Microsoft.Bot.Builder.Dialogs.ChoicePrompt.OnRecognizeAsync(ITurnContext turnContext, IDictionary`2 state, PromptOptions options, CancellationToken cancellationToken) in D:\a\1\s\libraries\Microsoft.Bot.Builder.Dialogs\Prompts\ChoicePrompt.cs:line 97
   at Microsoft.Bot.Builder.Dialogs.Prompt`1.<ContinueDialogAsync>d__5.MoveNext() in D:\a\1\s\libraries\Microsoft.Bot.Builder.Dialogs\Prompts\Prompt.cs:line 82
```

## To Reproduce
1. Test a bot that uses both adaptive submit actions and choice prompts (the submit action should not have a string for its `Data` property which would simulate `imBack` behavior instead of `postBack`)
2. When prompted to make a choice, click on an adaptive card's submit button
3. See error

## Expected behavior
The SDK should be able to handle textless messages gracefully. A null or empty `Text` property should be treated the same as white space. In the case of a choice prompt, the SDK should respond the same way to a textless message as it would if any other message was received that didn't contain a recognizable choice. In the case of LUIS recognizers, a textless message should probably be recognized as a `None` intent.

While it may seem acceptable for a textless message to throw an error in these situations because it means the user is doing something they shouldn't, consider that some developers are trying to incorporate cards into prompts, like in this case: https://github.com/Microsoft/botbuilder-dotnet/issues/1202

## Screenshots
![null utterance](https://user-images.githubusercontent.com/41968495/49409442-8c98d400-f715-11e8-85cd-4bb83f0c1d3a.jpg)

## Additional context
This is related to this issue which was considered fixed: https://github.com/Microsoft/botbuilder-dotnet/issues/572

Because the error seems to be showing up in multiple places, I don't want to limit the scope to just one spot in the code. I'm hoping some developers who are more familiar with the code base can implement a solution that applies to every instance of this pattern. The places I've found it are in [Find.cs](https://github.com/Microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Builder.Dialogs/Choices/Find.cs#L24) and [LuisRecognizer.cs](https://github.com/Microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs#L416).

[bug]

**Comment by hansmbakker** [comment]
@stevengum @v-kydela can you please reopen this?

This is not fixed only by fixing choicerecognizer: if you click a button in an Adaptive Card that does a `postBack` then it sends a `message` activity with a `value` property and without text. This causes errors in Virtual Assistant at the moment.

**Comment by EricDahlvang** [comment]
Hi @hansmbakker 

The community libraries provide an AdaptiveCardPrompt which works with Adaptive Cards:

https://github.com/BotBuilderCommunity/botbuilder-community-dotnet/blob/develop/libraries/Bot.Builder.Community.Dialogs.Prompts/AdaptiveCardPrompt.cs

nuget:
https://www.nuget.org/packages/Bot.Builder.Community.Dialogs.Prompts/

If you find this does not meet your needs, please open a new issue.

**Comment by hansmbakker** [comment]
Hi @EricDahlvang,
thank you for the suggestion! Does that cover the use case where the button in the card can trigger an optional action, that the user can also trigger a few turns later?

When handling it as a prompt, I suppose that the prompt listens for it only once (as a response to the Adaptive Card message from the bot to the user)?
